lib/rmt/cli/systems.rb

```diff
@@ -5,20 +5,38 @@ class RMT::CLI::Systems < RMT::CLI::Base
   option :csv, type: :boolean, desc: _('Output data in CSV format')
 
   def list
-    systems = (options.all ? System.all : System.limit(options.limit)).order(id: :desc)
-    decorator = RMT::CLI::Decorators::SystemDecorator.new(systems)
+    systems = System.limit(options.limit).order(id: :desc)
+    decorator = RMT::CLI::Decorators::SystemDecorator.new(systems) unless options.all
 
     if systems.empty?
       warn _('There are no systems registered to this RMT instance.')
     elsif options.csv
-      puts decorator.to_csv
+      if options.all
+        decorator = RMT::CLI::Decorators::SystemDecorator.new(systems.first, all: true)
+        puts decorator.csv_headers
+        System.in_batches(order: :desc, load: true) do |systems|
+          systems.each do |system|
+            decorator = RMT::CLI::Decorators::SystemDecorator.new(system, all: true)
+            puts decorator.to_csv(batch: true)
+          end
+        end
```

Comment:
Since you are using batches, are there any specific needs which justify instantiating a single decorator for each system?

As I can understand, you are already querying the system in batches. Wouldn't the batch size be OK to render at once all systems in the batch?

Then, you could remove the `all` flag, which is a bit misleading both at this point and in the decorator class, since it is called "all" but handles just a single instance.